server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DomainRoutes.java

```diff
@@ -59,17 +77,73 @@ public DomainRoutes(DomainList domainList, JsonTransformer jsonTransformer) {
 
     @Override
     public void define(Service service) {
-        service.get(DOMAINS,
-            (request, response) -> domainList.getDomains(),
-            jsonTransformer);
+        this.service = service;
 
-        service.get(SPECIFIC_DOMAIN, this::exists);
+        getDomains();
 
-        service.put(SPECIFIC_DOMAIN, this::addDomain);
+        domainIsExisted();
+
+        addDomain();
 
+        deleteDomain();
```

Comment:
maybe you could prefix these method by "define", because like this we can think you are effectively deleting a domian. So that would be defineDeleteDomain, defineGetDomains, etc.